src/libhirte/common/config.c

```diff
@@ -0,0 +1,180 @@
+                if (result != 0) {
+                        return result;
+                }
+        }
+
+        if (custom_config_file != NULL) {
+                result = config_load_from_file(custom_config_file);
+                // if custom config file doesn't exist, continue
+                if (result != 0 && result != ENOENT) {
+                        return result;
+                }
+        }
+
+        if (custom_config_directory != NULL) {
+                // TODO: https://github.com/containers/hirte/issues/148
+        }
+
+        result = config_load_from_env();
+
+        return result;
+}
+
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+/*
+ * Handle used to store option parsed from a file into a configuration.
+ *
+ * Returns 1 on success, 0 on error as per ini library requirements.
+ */
+static int parsing_handler(void *user, const char *section, const char *name, const char *value) { // NOLINT
+        if (config_set_value(name, value) != 0) {
+                return 0;
+        }
+        return 1;
+}
+#pragma GCC diagnostic pop
+
+int config_load_from_file(const char *config_file) {
+        assert(config_file != NULL);
+        if (access(config_file, F_OK) != 0) {
+                return ENOENT;
+        }
+        return ini_parse(config_file, parsing_handler, config);
+}
+
+int config_load_from_env() {
+        const char *env_vars[] = { ENV_LOG_LEVEL, ENV_LOG_TARGET, ENV_LOG_IS_QUIET };
+
+        const char *option_names[] = { CFG_LOG_LEVEL, CFG_LOG_TARGET, CFG_LOG_IS_QUIET };
+
+        int length = sizeof(env_vars) / sizeof(env_vars[0]);
+        for (int i = 0; i < length; i++) {
+                char *value = getenv(env_vars[i]);
+                if (value != NULL && strlen(value) > 0) {
+                        // environment variable is defined and has a non-empty value, so store it to the configuration
+                        int result = config_set_value(option_names[i], value);
+                        if (result != 0) {
+                                return result;
+                        }
+                }
+        }
+        return 0;
+}
+
+int config_set_value(const char *option_name, const char *option_value) {
+        char *name_copy = strdup(option_name);
+        if (name_copy == NULL) {
+                return ENOMEM;
+        }
+
+        char *value_copy = (char *) option_value;
+        if (value_copy != NULL) {
+                if (strlen(value_copy) > 0) {
+                        // storing non NULL value, so copy is needed
+                        value_copy = strdup(option_value);
+                        if (value_copy == NULL) {
+                                free(name_copy);
+                                return ENOMEM;
+                        }
+                } else {
+                        // empty string mean non-existent value, so store NULL
+                        value_copy = NULL;
+                }
+        }
```

Comment:
I don't like this cast. Not sure why the compiler allowed it. I think this should be better:

```suggestion
        char *value_copy = NULL;
        if (option_value != NULL && strlen(option_value) > 0) {
            // storing non NULL value, so copy is needed
            value_copy = strdup(option_value);
            if (value_copy == NULL) {
                free(name_copy);
                return ENOMEM;
            }
        }
```

Reply:
Fixed